internal/patch/add.go

```diff
@@ -0,0 +1,155 @@
+package patch
+
+import (
+	"fmt"
+	f "github.com/elimity-com/scim/internal/filter"
+	"github.com/elimity-com/scim/schema"
+	"github.com/scim2/filter-parser/v2"
+	"strings"
+)
+
+// getRefAttribute returns the corresponding attribute based on the given attribute path.
+//
+// e.g.
+//  - `userName` would return the userNameAttribute.
+//	- `name.givenName` would return the `givenName` attribute.
+//  - `ext:employeeNumber` would return the `employeeNumber` from the extension.
+func (v OperationValidator) getRefAttribute(attrPath filter.AttributePath) (*schema.CoreAttribute, error) {
+	// Get the corresponding schema, this can be the main schema or an extension.
+	var refSchema = v.schema
+	if uri := attrPath.URI(); uri != "" {
+		// It can also be an extension if it has a uri prefix.
+		var ok bool
+		if refSchema, ok = v.schemas[uri]; !ok {
+			return nil, fmt.Errorf("invalid uri prefix: %s", uri)
+		}
+	}
+
+	// Get the correct attribute corresponding to the given attribute path.
+	var (
+		refAttr  *schema.CoreAttribute
+		attrName = attrPath.AttributeName
+	)
+	for _, attr := range refSchema.Attributes {
+		if strings.EqualFold(attr.Name(), attrName) {
+			refAttr = &attr
+			break
+		}
+	}
+	if refAttr == nil {
+		// This should not occur since the filter parser will validate this.
+		return nil, fmt.Errorf("could not find attribute %s", v.path)
```

Comment:
If this should not occur, you can just leave the check out IMO.

Reply:
After re-evaluation this check is needed since `getRefAttribute` also is used on paths in the `value` field, which is not validated by the filter package.